### ISSUE TITLE
ERROR: Destroying WAF Resources 

### ISSUE THREAD

**Issue by tstrohmeier** [opentelekomcloud/terraform-provider-opentelekomcloud#372]
Hi there,

### Affected Resource(s)
Please list the resources as a list, for example:
- opentelekomcloud_waf_domain_v1
- opentelekomcloud_waf_policy_v1

### Terraform Configuration Files
```hcl

resource "opentelekomcloud_waf_domain_v1" "domain_1" {

  hostname = "domain.com"

   ...

  certificate_id ="1234"
}

resource "opentelekomcloud_waf_policy_v1" "policy_1" {
  name = "policy-1"
  level = 2
  hosts = [opentelekomcloud_waf_domain_v1.domain_1.id]
 
 ...

}

```

### Debug Output
```
opentelekomcloud_waf_policy_v1.policy_1: Destroying... [id=xx]

Error: Error deleting OpenTelekomCloud WAF Policy: Bad request with: [DELETE https://waf.eu-de.otc.t-systems.com/v1/xxx/waf/policy/xxx], error message: {"error_msg": "Failed to delete the policy because it is being used","error_code":"WAF.3003"}
```

### Expected Behavior
The provider should delete the domain first and afterwards the policy

### Actual Behavior
The provider tried to delete the policy first.

### Steps to Reproduce
1. `terraform plan -destroy` or `terraform destroy`

### References
- #303
- @Guchelkaben
- @niuzhenguo

**Comment by tstrohmeier** [comment]
We took a look on the UI. It is possible to assign multible domains to one policy, but a domain is only allowed to have one policy. 

`domain n:1 policy` 

@niuzhenguo I would suggest to specify the policy id on the domain resource.

The modified example:

```
resource "opentelekomcloud_waf_domain_v1" "domain_1" {

  hostname = "domain.com"

  policy = opentelekomcloud_waf_policy_v1.policy_1.id
   ...

  certificate_id ="1234"
}

resource "opentelekomcloud_waf_policy_v1" "policy_1" {
  name = "policy-1"
  level = 2
 ...

}
```

--- 

Workaround we used now to get around this limitation:
1. Comment out the policy resource
2. Destroy the domain `terraform destroy -target opentelekomcloud_waf_domain_v1.domain_1 `
